### PR TITLE
Added multi selection support for scalable settings in lights

### PR DESCRIPTION
**Pull request by fredericv-unity3d**
### Purpose of this PR
Fixes [1182968](https://fogbugz.unity3d.com/f/cases/1182968/)

Refactored ScalableSettings to support different number of level per setting.
Added support for multi selection

---
### Release Notes
Fixes [1182968](https://fogbugz.unity3d.com/f/cases/1182968/)

---
### Testing status
**Katana Tests**: -

**Automated Tests**: -

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: Low

---

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HDEditorUtils.cs [hunk 1]
```diff
@@ -294,5 +295,162 @@ public static T GetEnumValue<T>(this SerializedProperty property)
...
+        /// </summary>
+        /// <typeparam name="T">
+        /// The type of the value to get.
+        ///
+        /// It is expected to be a supported type by the <see cref="SerializedProperty"/>.
+        /// </typeparam>
+        /// <param name="serializedProperty">The property to get.</param>
+        /// <returns>The value of the property.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static T GetInline<T>(this SerializedProperty serializedProperty)
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HDEditorUtils.cs]: Usefull, I'll need to clean FrameSettings that use the same scheme when this land. Thanks for exposing it in a library.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs [hunk 2]
```diff
@@ -423,7 +423,7 @@ partial struct FrameSettings
         public LODBiasMode lodBiasMode;
         /// <summary>The quality level to use when fetching the quality setting value.</summary>
         [SerializeField]
-        public ScalableSetting.Level lodBiasQualityLevel;
+        public int lodBiasQualityLevel;
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs]: Can this be type changed without migration?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs [hunk 3]
```diff
@@ -435,7 +435,7 @@ partial struct FrameSettings
         public MaximumLODLevelMode maximumLODLevelMode;
         /// <summary>The quality level to use when fetching the quality setting value.</summary>
         [SerializeField]
-        public ScalableSetting.Level maximumLODLevelQualityLevel;
+        public int maximumLODLevelQualityLevel;
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs]: same

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/ScalableSetting.cs [hunk 4]
```diff
@@ -1,117 +1,121 @@
...
 
-            dst.m_Level = m_Level;
-            dst.m_UseOverride = m_UseOverride;
-            dst.m_Override = m_Override;
-        }
+    /// <summary><see cref="ScalableSetting{T}"/>.</summary>
+    [Serializable]
+    public class IntScalableSetting : ScalableSetting<int>
+    {
+        public IntScalableSetting(int[] values, ScalableSettingSchemaId schemaId) : base(values, schemaId) { }
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/ScalableSetting.cs]: Can't this constructor be declared in the ScalableSetting<T> class instead? It seams redundant to redefine the same for each of the following classes with only type varying when base classe is generic...

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs (hunk 2 above)
Comment (fredericv-unity3d) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs]: As far as I know for the serialization framework enum can be aliased as int.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/ScalableSetting.cs (hunk 4 above)
Comment (fredericv-unity3d) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/ScalableSetting.cs]: C# inheritance constraint

**Pull request merged**